bentoml/deployment/utils.py

```diff
@@ -44,3 +52,77 @@ def process_docker_api_line(payload):
                     )
                 elif "stream" in line_payload:
                     logger.info(line_payload['stream'])
+
+
+INSTALL_TARGZ_TEMPLATE = """\
+#!/bin/bash
+
+for filename in ./bundled_dependencies/*.tar.gz; do
+    [ -e "$filename" ] || continue
+    pip install "$filename" --ignore-installed
+done
+"""
+
+
+def add_local_bentoml_package_to_repo(archive_path):
+    try:
+        module_location, = importlib.util.find_spec('bentoml').submodule_search_locations
+    except AttributeError:
+        # python 2.7 doesn't have importlib.util, will fall back to imp instead
+        import imp
+        _, module_location, _ = imp.find_module('bentoml')
+    bentoml_setup_py = os.path.abspath(os.path.join(module_location, '..', 'setup.py'))
+
+    if not os.path.isfile(bentoml_setup_py):
+        raise KeyError('"setup.py" for Bentoml module not found')
+
+    # Create random directory inside bentoml module for storing the bundled
+    # targz file. Since dist-dir can only be inside of the module directory
+    date_string = datetime.now().strftime("%Y_%m_%d")
+    bundle_dir_name = '__bento_dev_{}'.format(date_string)
+    Path(bundle_dir_name).mkdir(exist_ok=True, parents=True)
+
+    sandbox.run_setup(
+        bentoml_setup_py, ['sdist', '--format', 'gztar', '--dist-dir', bundle_dir_name]
+    )
+
+    # copy the generated targz to archive directory and remove it from
+    # bentoml module directory
+    source_dir = os.path.abspath(os.path.join(module_location, '..', bundle_dir_name))
+    dest_dir = os.path.join(archive_path, 'bundled_dependencies')
+    if os.path.isdir(source_dir):
+        shutil.copytree(source_dir, dest_dir)
+        shutil.rmtree(source_dir)
+
+    # Include script for install targz file in archive directory
+    install_script_path = os.path.join(archive_path, 'install_bundled_dependencies.sh')
```

Comment:
let's move this part to `save_to_dir` method and make the script always saved in a Bento directory along with the dockerfile -- so, later on, we can add an API allow users to add their custom python libraries this way as well

and let's rename it to `bentoml_init.sh` so we need to add other bentoML internal staff that's required when initializing a container